chaos_monkey/network/network_scanner.py

```diff
@@ -27,48 +52,72 @@ def initialize(self):
                             for ip_address in self._ip_addresses]
         if WormConfiguration.local_network_scan:
             self._ranges += [FixedRange([ip_address for ip_address in get_ips_from_interfaces()])]
         LOG.info("Base local networks to scan are: %r", self._ranges)
 
-    def get_victim_machines(self, scan_type, max_find=5, stop_callback=None):
-        assert issubclass(scan_type, HostScanner)
+    def get_victim_machines(self, scan_type, max_find=5, scan_size=ITERATION_BLOCK_SIZE, stop_callback=None):
+        """
+        Generator of victim machines based on the network scanning configuration.
+        Every time scans
+        :param scan_type: scanner_class type
+        :param max_find:  Maximum number of victims to find
+        :param scan_size: Number of hosts to scan in parralel
+        :param stop_callback: Callback per iteration to check if we should stop activity
+        :return: Generates VictimHost instances
+        """
 
         scanner = scan_type()
         victims_count = 0
+        pool = ThreadPool(ITERATION_BLOCK_SIZE)
 
         for net_range in self._ranges:
             LOG.debug("Scanning for potential victims in the network %r", net_range)
-            for victim in net_range:
+            for scan_chunk in _grouper(net_range, scan_size):
                 if stop_callback and stop_callback():
                     LOG.debug("Got stop signal")
                     break
 
                 # skip self IP address
-                if victim.ip_addr in self._ip_addresses:
-                    continue
-
+                scan_chunk = [x for x in scan_chunk if x.ip_addr not in self._ip_addresses]
                 # skip IPs marked as blocked
-                if victim.ip_addr in WormConfiguration.blocked_ips:
+                bad_victims = [x for x in scan_chunk if x.ip_addr in WormConfiguration.blocked_ips]
+                for victim in bad_victims:
                     LOG.info("Skipping %s due to blacklist" % victim)
-                    continue
 
-                LOG.debug("Scanning %r...", victim)
+                scan_chunk = [x for x in scan_chunk if x.ip_addr not in WormConfiguration.blocked_ips]
 
-                # if scanner detect machine is up, add it to victims list
-                if scanner.is_host_alive(victim):
-                    LOG.debug("Found potential victim: %r", victim)
+                LOG.debug("Scanning %r...", scan_chunk)
+
+                results = pool.map(partial(self.scan_machine, scanner=scanner),
+                                   scan_chunk)
+                victims_chunk = [x for x in results if x]
+
+                for victim in victims_chunk:
                     victims_count += 1
                     yield victim
-
                     if victims_count >= max_find:
                         LOG.debug("Found max needed victims (%d), stopping scan", max_find)
+                        return
 
-                        break
 
                 if SCAN_DELAY:
```

Comment:
PEP 8: too much blank lines

Reply:
Fixed, not sure why it made it to the commit.